infra/bicep/core/ai/ai-hub-secure.bicep

```diff
@@ -42,17 +42,36 @@ param addRoleAssignments bool = true
 param userObjectId string = ''
 param userObjectType string = 'User'
 
-@description('The object ID of the application identity to be granted necessary role assignments to access the Azure AI Hub.')
-param managedIdentityId string = ''
+// @description('The object ID of the application identity to be granted necessary role assignments to access the Azure AI Hub.')
+// param managedIdentityResourceId string = ''
+@description('The principal ID of the application identity to be granted necessary role assignments to access the Azure AI Hub.')
+param managedIdentityPrincipalId string = ''
 param managedIdentityType string = 'ServicePrincipal'
 
-resource aiHub 'Microsoft.MachineLearningServices/workspaces@2024-04-01-preview' = {
+@description('Name AI Search resource')
+param aiSearchName string
+param aiSearchResourceGroupName string = resourceGroup().name
+
+var aiServiceConnectionName = '${aiHubName}-connection-AIService'
+var searchConnectionName  = '${aiHubName}-connection-AISearch'
+
+resource aiSearch 'Microsoft.Search/searchServices@2020-03-13' existing = {
+  name: aiSearchName
+  scope: resourceGroup(aiSearchResourceGroupName)
+}
+
+resource aiHub 'Microsoft.MachineLearningServices/workspaces@2024-10-01' = {
```

Comment:
agents capability is missing and doesn't work after deployment
![image](https://github.com/user-attachments/assets/4522e5ed-dd2f-4a7d-b305-abe84100fbd2)

Reply:
I just deployed a fresh one and it has the agents capability:
![image](https://github.com/user-attachments/assets/f7909b51-dc2b-4dba-a3f0-7badbef01028)

Reply:
what happens if you do "let's go" ?